demo/admin/src/products/future/generated/CreateCapProductForm.tsx

```diff
@@ -15,35 +15,39 @@ import {
     useStackSwitchApi,
 } from "@comet/admin";
 import { FinalFormDatePicker } from "@comet/admin-date-time";
-import { Lock } from "@comet/admin-icons";
 import { BlockState, createFinalFormBlock } from "@comet/blocks-admin";
 import { DamImageBlock, EditPageLayout } from "@comet/cms-admin";
-import { FormControlLabel, InputAdornment, MenuItem } from "@mui/material";
+import { FormControlLabel } from "@mui/material";
+import { GQLProductType } from "@src/graphql.generated";
 import { FormApi } from "final-form";
 import isEqual from "lodash.isequal";
 import React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { validateTitle } from "../validateTitle";
-import { createProductMutation, productCategoriesQuery } from "./CreateProductForm.gql";
+import { createProductMutation, productCategoriesQuery } from "./CreateCapProductForm.gql";
 import {
+    GQLCreateCapProductFormDetailsFragment,
     GQLCreateProductMutation,
     GQLCreateProductMutationVariables,
     GQLProductCategoriesSelectQuery,
     GQLProductCategoriesSelectQueryVariables,
     GQLProductCategorySelectFragment,
-    GQLProductFormDetailsFragment,
-} from "./CreateProductForm.gql.generated";
+} from "./CreateCapProductForm.gql.generated";
 
 const rootBlocks = {
     image: DamImageBlock,
 };
 
-type FormValues = GQLProductFormDetailsFragment & {
+type FormValues = GQLCreateCapProductFormDetailsFragment & {
     image: BlockState<typeof rootBlocks.image>;
 };
 
-export function CreateProductForm(): React.ReactElement {
+interface FormProps {
+    type: GQLProductType;
+}
+
+export function CreateCapProductForm({ type }: FormProps): React.ReactElement {
```

Comment:
we need more demo-code. I didn't considered if the form is also used to edit the entity. in this case this additional props are not required, because they only apply for create. so this props should require id or required props... @nsams @johnnyomair should I address this in another pull-request?

Reply:
are you thinking of `{ id: string } | { type: GQLProductType }`? Does this even work?

Reply:
it seems this is possible. I found an article describing the solution is defining never for the other props. But this does only validate the type does not match but does not help with autocomplete.

```
type FormProps =
    | {
        id: string;
        type?: never;
      }
    | {
        id?: never;
        type: GQLProductType;
      };
```

my other solution would be not working with root-props but with nested

```
type FormProps = {
    data: {
        id: string;
    } | {
        type: GQLProductType;
    }
}
```